File: HyperFetchMobile/app/src/main/java/com/hyperfetch/model/TaskStatus.java
package com.hyperfetch.model;

/**
 * 任务状态枚举
 * 用于标识下载任务的当前状态
 */
public enum TaskStatus {
    /**
     * 排队中 - 任务已创建，等待开始下载
     */
    QUEUED,

    /**
     * 下载中 - 任务正在下载
     */
    DOWNLOADING,

    /**
     * 已暂停 - 任务被用户暂停
     */
    PAUSED,

    /**
     * 已完成 - 任务下载完成
     */
    COMPLETED,

    /**
     * 失败 - 任务下载失败
     */
    FAILED,

    /**
     * 完成但有警告 - 任务完成但存在问题（如合并失败保留原始文件）
     */
    COMPLETED_WITH_WARNING
}

File: common/src/main/java/com/downloadmanager/common/TaskStatus.kt
package com.downloadmanager.common

enum class TaskStatus(val displayName: String) {
    WAITING("等待中"),
    QUEUED("排队中"),
    DOWNLOADING("下载中"),
    MERGING("合并中"),
    COMPLETED("已完成"),
    PAUSED("已暂停"),
    FAILED("失败"),
    CANCELLED("已取消");

    fun canTransitionTo(target: TaskStatus): Boolean {
        return when (this) {
            WAITING -> target in setOf(QUEUED, DOWNLOADING, CANCELLED)
            QUEUED -> target in setOf(DOWNLOADING, PAUSED, CANCELLED)
            DOWNLOADING -> target in setOf(MERGING, PAUSED, FAILED, CANCELLED)
            MERGING -> target in setOf(COMPLETED, FAILED)
            PAUSED -> target in setOf(DOWNLOADING, CANCELLED)
            FAILED -> target in setOf(WAITING, CANCELLED)
            COMPLETED -> false
            CANCELLED -> false
        }
    }

    val isTerminal: Boolean get() = this == COMPLETED || this == CANCELLED
    val isActive: Boolean get() = this == DOWNLOADING || this == MERGING
    val isWaiting: Boolean get() = this == WAITING || this == QUEUED
}